File: Calto-GG/Goldman Sachs/Day 1/Valid_Square.py

# Just check square properties for side and diagonal
class Solution:
    def validSquare(self, p1: List[int], p2: List[int], p3: List[int], p4: List[int]) -> bool:
        def dist(point1,point2):
            return (point1[0]-point2[0])**2+(point1[1]-point2[1])**2
            
        D=[dist(p1,p2),dist(p1,p3),dist(p1,p4),dist(p2,p3),dist(p2,p4),dist(p3,p4)]
        D.sort()
        return 0<D[0]==D[1]==D[2]==D[3] and D[4]==D[5]

File: nilay302/Goldman sachs/Day 6/Valid Square.cpp
class Solution {
public:
    bool validSquare(vector<int>& p1, vector<int>& p2, vector<int>& p3, vector<int>& p4) {
        vector<int>dist;
        dist.push_back((p1[0]-p2[0])*(p1[0]-p2[0]) + (p1[1]-p2[1])*(p1[1]-p2[1]));
        dist.push_back((p2[0]-p3[0])*(p2[0]-p3[0]) + (p2[1]-p3[1])*(p2[1]-p3[1]));
        dist.push_back((p1[0]-p3[0])*(p1[0]-p3[0]) + (p1[1]-p3[1])*(p1[1]-p3[1]));
        dist.push_back((p1[0]-p4[0])*(p1[0]-p4[0]) + (p1[1]-p4[1])*(p1[1]-p4[1]));
        dist.push_back((p3[0]-p4[0])*(p3[0]-p4[0]) + (p3[1]-p4[1])*(p3[1]-p4[1]));
        dist.push_back((p4[0]-p2[0])*(p4[0]-p2[0]) + (p4[1]-p2[1])*(p4[1]-p2[1]));

        sort(dist.begin(),dist.end());

        for(int i = 0;i<dist.size();i++){
            if(dist[i]==0)return false;
        }

        if(dist[0]==dist[1] && dist[1]==dist[2] && dist[2] == dist[3]  && dist[4]==dist[5])return true;

        return false;
    }
};